src/Sarif/Core/Run.cs

```diff
@@ -13,6 +14,109 @@ public partial class Run
         private static Invocation EmptyInvocation = new Invocation();
         private static LogicalLocation EmptyLogicalLocation = new LogicalLocation();
 
+        private IDictionary<FileLocation, int> _fileToIndexMap;
+
+        public int GetFileIndex(
+            FileLocation fileLocation, 
+            bool addToFilesTableIfNotPresent = true,
+            OptionallyEmittedData dataToInsert = OptionallyEmittedData.None,
+            System.Text.Encoding encoding = null)
+        {
+            if (fileLocation == null) throw new ArgumentNullException(nameof(fileLocation));
```

Comment:
>throw [](start = 38, length = 5)

I am absolutely fine with this construct, but I think you've told me before that you prefer braces even if it's a single statement on the same line.

Reply:
it's not my preference, it's my attempt to create a standard that balances the preferences of many. 

i.e., i do not set the coding guidelines in this code base according to my own preferences but instead attempt to help negotiate a set of standards that usefully encompasses valid concerns raised by the range of developers who contribute.

---
In reply to: [244521685](https://github.com/Microsoft/sarif-sdk/pull/1191#discussion_r244521685) [](ancestors = 244521685)